dataRetriever/blockchain/baseBlockchain.go

```diff
@@ -70,3 +71,25 @@ func (bbc *baseBlockChain) SetCurrentBlockHeaderHash(hash []byte) {
 	bbc.currentBlockHeaderHash = hash
 	bbc.mut.Unlock()
 }
+
+func (bbc *baseBlockChain) SetHighestFinalBlockCoordinates(nonce uint64, headerHash []byte, rootHash []byte) {
```

Comment:
Missing comments.
I would not call these "coordinates"
Let's rename to SetHighestFinalBlockInfo(...)